docs/resources/azure_data_lake_storage_gen2_filesystems.md

```diff
@@ -0,0 +1,98 @@
+---
+title: About the azure_data_lake_storage_gen2_filesystems Resource
+platform: azure
+---
+
+# azure_data_lake_storage_gen2_filesystems
+
+Use the `azure_data_lake_storage_gen2_filesystems` InSpec audit resource to test the properties related to all Azure Data Lake Storage Gen2 Filesystems within a project.
+
+## Azure REST API version, Endpoint, and HTTP Client Parameters
+
+This resource interacts with API versions supported by the resource provider. The `api_version` is defined as a resource parameter.
+If not provided, the latest version is used. For more information, refer to [`azure_generic_resource`](azure_generic_resource.md).
+
+Unless defined, `azure_cloud` global endpoint and default values for the HTTP client is used. For more information, refer to the resource pack [README](../../README.md).
+
+## Availability
+
+### Installation
+
+This resource is available in the [InSpec Azure resource pack](https://github.com/inspec/inspec-azure). For an example, `inspec.yml` file and how to set up your Azure credentials, refer to resource pack [README](../../README.md#Service-Principal).
+
+## Syntax
+
+An `azure_data_lake_storage_gen2_filesystems` resource block returns all Azure Data Lake Storage Gen2 Filesystems within a project.
+
+```ruby
+describe azure_data_lake_storage_gen2_filesystems(account_name: 'ACCOUNT_NAME') do
+  #...
+end
+```
+
+## Parameters
+| Name           | Description                                                                      |
+|----------------|----------------------------------------------------------------------------------|
+| account_name   | The Azure Storage account name.                                                  |
+| dns_suffix     | The DNS suffix for the Azure Data Lake Storage endpoint.                         |
```

Comment:
Can you just confirm .If user want to provide dns_suffix ,How he will do that ?

Reply:
yeah so the `dns_suffix` could also be a custom suffix, that is why we need provide an optional way to accept it